packages/database/src/migrations/20210511024543-CreateDashboardItemsTableLayout-modifies-schema.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+exports.up = async function (db) {
+  await db.runSql(`
+    CREATE TABLE dashboard_item (
+      id TEXT PRIMARY KEY,
+      code TEXT NOT NULL UNIQUE,
+      config JSONB NOT NULL DEFAULT '{}',
+      permission_groups TEXT[] NOT NULL CONSTRAINT permission_groups_not_empty CHECK (permission_groups <> '{}'),
+      report_code TEXT NOT NULL,
+      FOREIGN KEY (report_code) REFERENCES report (code) ON UPDATE CASCADE ON DELETE RESTRICT
```

Comment:
Self admittedly these column constraints are perhaps overly restrictive, I'm not sure if we have a particular way of deciding on column constraints, but this is built around my understanding of the intended usage